libraries/plugins/apis/database_api/database_api.cpp

```diff
@@ -1425,40 +1425,94 @@ DEFINE_API_IMPL( database_api_impl, verify_signatures )
 
 DEFINE_API_IMPL( database_api_impl, find_transaction )
 {
-    // Yet to check the argument list
-    // FC_ASSERT ( args.size() == 1 || args.size() == 2, "Expected 1-2 arguments, was ${n}", ("n", args.size()) );
-    // auto trx_id = { args[0].as< transaction_id_type >() };
-    // auto expiration_time = { args[1].as< time_point_sec >() };
+    FC_ASSERT ( args.size() == 1 || args.size() == 2, "Expected 1-2 arguments, was ${n}", ("n", args.size()) );
     
     find_transaction_return result;
-    //transaction_status_codes result;
-    result.trx_status_code = transaction_status_codes::not_aware_of_trx; // 0 - Not aware of the transaction
     
-    auto trx_id = args.trx_id.as<transaction_id_type>();
+    time_point_sec expiration;
+    transaction_id_type id = { args[0].as< transaction_id_type >() };
+    const auto& trx_idx = _db.get_index<transaction_index>().indices().get<by_trx_id>();
+    const auto itr_idx_tx = trx_idx.find(id);
     
-    // Only transaction id is provided
-    //if (args.size() == 1)
+    const auto itr_pending_tx = std::find_if(_db._pending_tx.begin(), _db._pending_tx.end(),
+                                    [&](const signed_transaction& t) {
+                                        return t.id() == id;
+                                    });
+    bool trx_in_mempool = itr_pending_tx != _db._pending_tx.end() ? true : false;
+    bool trx_in_index = itr_idx_tx != trx_idx.end() ? true : false;
+    
+    switch (args.size())
     {
-        try {
-            if (_db.is_known_transaction(trx_id))
+        case 1:
+            // 0 - Not aware of the transaction
+            if (!trx_in_mempool && !trx_in_index)
```

Comment:
`trx_in_mempool == true` implies `trx_in_index == true` because the current state is the head block plus all pending transactions, which means they will exist in the index.